### PR TITLE
Move Podspecs to v1

### PR DESCRIPTION
**Pull request by jcanizales**
Analogous to https://github.com/grpc/grpc/pull/7413

Expected Travis failure of the "build examples" tests until the podspecs are pushed. And failures of the other tests until the BoringSSL podspec is pushed. I did test them by temporarily modifying `BoringSSL.podspec` to point to a commit instead of a tag.

### CODE REVIEW

**Comment by muxi** [comment]
src/objective-c/GRPCClient/private/GRPCHost.m:52 has another version number that I think needs to modify

File: examples/objective-c/auth_sample/AuthTestService.podspec [hunk 1]
```diff
@@ -14,7 +14,7 @@ Pod::Spec.new do |s|
   src = "../../protos"
 
   # Run protoc with the Objective-C and gRPC plugins to generate protocol messages and gRPC clients.
-  s.dependency "!ProtoCompiler-gRPCPlugin", "~> 1.0.0-pre1.2"
+  s.dependency "!ProtoCompiler-gRPCPlugin", "~> 1.0"
```
Comment (muxi) [inline examples/objective-c/auth_sample/AuthTestService.podspec]: Should this be 1.0 or 1.0.0? :)

File: src/objective-c/!ProtoCompiler-gRPCPlugin.podspec [hunk 2]
```diff
@@ -82,10 +82,9 @@ Pod::Spec.new do |s|
   s.authors  = { 'The gRPC contributors' => 'grpc-packages@google.com' }
 
   repo = 'grpc/grpc'
-  release = "objective-c-v#{v}"
   file = "grpc_objective_c_plugin-#{v}-macos-x86_64.zip"
   s.source = {
-    :http => "https://github.com/#{repo}/releases/download/#{release}/#{file}",
+    :http => "https://github.com/#{repo}/releases/download/v#{v}/#{file}",
```
Comment (muxi) [inline src/objective-c/!ProtoCompiler-gRPCPlugin.podspec]: I got 404 downloading this file. Can you confirm?

File: src/objective-c/examples/RemoteTestClient/RemoteTest.podspec [hunk 3]
```diff
@@ -11,7 +11,7 @@ Pod::Spec.new do |s|
   s.osx.deployment_target = '10.9'
 
   # Run protoc with the Objective-C and gRPC plugins to generate protocol messages and gRPC clients.
-  s.dependency "!ProtoCompiler-gRPCPlugin", "~> 1.0.0-pre1.2"
+  s.dependency "!ProtoCompiler-gRPCPlugin"
```
Comment (muxi) [inline src/objective-c/examples/RemoteTestClient/RemoteTest.podspec]: Do we need version number here?

File: src/objective-c/!ProtoCompiler-gRPCPlugin.podspec (hunk 2 above)
Comment (jcanizales) [inline src/objective-c/!ProtoCompiler-gRPCPlugin.podspec]: When we create the release notes here https://github.com/grpc/grpc/releases we'll have to add this zip file (named `grpc_objective_c_plugin-1.0.0-macos-x86_64.zip`). @nicolasnoble, you'll create the release notes, right?

Comment (nicolasnoble) [inline src/objective-c/!ProtoCompiler-gRPCPlugin.podspec]: Yes. Don't forget to review them internally :)

File: src/objective-c/examples/RemoteTestClient/RemoteTest.podspec (hunk 3 above)
Comment (jcanizales) [inline src/objective-c/examples/RemoteTestClient/RemoteTest.podspec]: As this and `../../tests/RemoteTestClient/RemoteTest.podspec` are used only for testing, the `Podfile` always says to fetch the podspec from the local directory (whatever version that is). So specifying a version here is only giving us more work.

File: examples/objective-c/auth_sample/AuthTestService.podspec (hunk 1 above)
Comment (muxi) [inline examples/objective-c/auth_sample/AuthTestService.podspec]: That makes sense. Thanks~

**Comment by jcanizales** [comment]
BoringSSL 6.0 is now pushed to Cocoapods, and the required Jenkins tests pass :octocat: :octocat:

Merge?

**Pull request merged**